libraries/botframework-connector/src/auth/microsoftAppCredentials.ts

```diff
@@ -112,49 +111,37 @@ export class MicrosoftAppCredentials implements msrest.ServiceClientCredentials
         // 1. The user requested it via the forceRefresh parameter
         // 2. We have it, but it's expired
         // 3. We don't have it in the cache.
-        const newOAuthToken: OAuthResponse = await this.refreshToken();
-        MicrosoftAppCredentials.cache.set(this.tokenCacheKey, newOAuthToken);
+        const res: Response = await this.refreshToken();
+
+        let oauthResponse: OAuthResponse;
+        if (res.ok) {
+            oauthResponse = await res.json();
+            // Subtract 5 minutes from expires_in so they'll we'll get a
+            // new token before it expires.
+            oauthResponse.expiration_time = new Date(Date.now() + (oauthResponse.expires_in * 1000) - 300000);
```

Comment:
The type of `OAuthResponse.expiration_time` is a `Date`; if I use the new code it throws a type error as the right side evaluates to a number. I can change the `OAuthResponse` interface to use a number instead of Date for `expiration_time` with only one additional [change](https://github.com/Microsoft/botbuilder-js/pull/619/files#diff-2b849808f7c2abbd8344a3a41c788c74R104). 

The `OAuthResponse` interface is also not exported so there shouldn't be any problems with this change; what do you think?

Reply:
Either way is probably fine unless the `OAuthResponse` is stringified, then a numeric value makes sense since primitives exchange symmetrically with `JSON.stringify` and `JSON.parse`